deploy-agent/deployd/download/http_download_helper.py

```diff
@@ -65,3 +75,21 @@ def download(self, local_full_fn):
         except requests.ConnectionError:
             log.error('Could not connect to: {}'.format(self._url))
             return Status.FAILED
+
+    def validate_source(self):
+        tags = {'type': 'http', 'url': self._url}
+        create_sc_increment(DOWNLOAD_VALIDATE_METRICS, tags=tags)
+
+        parsed_url: ParseResult = urlparse(self._url)
+        if not parsed_url.scheme == 'https':
+            return False
+
+        domain: str = parsed_url.netloc
+        if not domain:
+            return False
+
+        allow_list = self._config.get_http_download_allow_list()
+        if allow_list is None:
+            return True
+
+        return domain in allow_list
```

Comment:
I wonder if we should use glob patterns here.

Reply:
we parse the url to get the domain and the domain list will be small <5 domains so glob patterns might not bring much benefit here.